app_dart/lib/src/request_handlers/push_gold_status_to_github.dart

```diff
@@ -108,31 +108,49 @@ class PushGoldStatusToGithub extends ApiRequestHandler<Body> {
 
       if (runsGoldenFileTests) {
         // Make sure we account for any running Luci builds
-        final List<String> luciIncompleteStates = <String>['failure', 'unreachable', 'pending'];
-        final List<String> luciStatuses = <String>[];
-        final Stream<RepositoryStatus> statusStream = gitHubClient.repositories.listStatuses(slug, pr.head.sha);
-        await for (RepositoryStatus status in statusStream) {
-          if (status.description != null && status.description.contains('LUCI')) {
-            log.debug('Found Luci build status for pull request #${pr.number}, '
-                'commit ${pr.head.sha}: ${status.description} (${status.state})');
-            luciStatuses.add(status.state);
+        final GraphQLClient gitHubGraphQLClient = await config.createGitHubGraphQLClient();
+        final List<String> luciIncompleteChecks = <String>[];
+        final Map<String, dynamic> data = await _queryGraphQL(
+          log,
+          gitHubGraphQLClient,
+          pr.number,
+        );
+        final Map<String, dynamic> prData = data['repository']['pullRequest'] as Map<String, dynamic>;
+        final Map<String, dynamic> commit = prData['commits']['nodes'].single['commit'] as Map<String, dynamic>;
+        List<Map<String, dynamic>> checkRuns;
+        if (commit['checkSuites']['nodes'] != null && (commit['checkSuites']['nodes'] as List<dynamic>).isNotEmpty) {
+          checkRuns = (commit['checkSuites']['nodes']?.first['checkRuns']['nodes'] as List<dynamic>)
+              .cast<Map<String, dynamic>>();
+        }
+        checkRuns = checkRuns ?? <Map<String, dynamic>>[];
+        log.debug('Found luci checks: $checkRuns');
+        for (Map<String, dynamic> checkRun in checkRuns) {
+          final String name = checkRun['name'] as String;
+          if (checkRun['status'] != null && checkRun['status'].toUpperCase() != 'COMPLETED') {
+            luciIncompleteChecks.add(name);
+          } else if (checkRun['conclusion'] != null && checkRun['conclusion'].toUpperCase() != 'SUCCESS') {
+            luciIncompleteChecks.add(name);
```

Comment:
Seems like most of this code is duplicated from check_for_waiting_pull_requests. Would it be possible to move it to github_checks_service and reuse it in both places?

Reply:
Definitely! Let me see what I can extract. The `check_for_waiting_pull_requests` queries for all PRs with the right label, while here we're looking up a specific PR. I might be able to just use the original query to get what I need.

Reply:
I think I'll open a separate PR to move the logic to `github_checks_service` and then update here.